zebra-network/src/peer_set/initialize.rs

```diff
@@ -182,47 +188,149 @@ where
+    let mut handshakes = FuturesUnordered::new();
+    // TODO: replace with *success_count_tx.borrow() in Tokio 1.6
+    let mut success_count = 0;
+    for candidate in initial_meta_addr {
+        // 1. Spawn a task to handshake with a new peer
+        let next_peer_sleep_until = Instant::now() + constants::MIN_PEER_CONNECTION_INTERVAL;
+        let hs_join = tokio::spawn(dial(candidate, outbound_connector.clone()))
+            .map(move |res| match res {
+                Ok(crawler_action) => crawler_action,
+                Err(e) => {
+                    panic!(
+                        "panic during initial handshake with {:?}: {:?} ",
+                        candidate, e
+                    );
+                }
+            })
+            .instrument(Span::current());
+        handshakes.push(Box::pin(hs_join));
+
+        // 2. Check if any peers have finished their handshakes
+        // If we've had enough successes, wait for all the handshakes to finish
+        while let Some(handshake_action) = if success_count < peerset_initial_target_size {
+            handshakes.next().now_or_never().flatten()
+        } else {
+            handshakes.next().await
+        } {
+            use CrawlerAction::*;
+            match handshake_action {
+                HandshakeConnected { peer_set_change } => {
+                    success_count += 1;
+                    if let Change::Insert(ref addr, _) = peer_set_change {
+                        debug!(
+                            ?addr,
+                            ?success_count,
+                            ?peerset_initial_target_size,
+                            "successfully dialed initial peer"
+                        );
+                    } else {
+                        unreachable!("unexpected handshake result: all changes should be Insert");
+                    }
+                    // the peer set is handled by an independent task, so this send
+                    // shouldn't hang
+                    peerset_tx.send(Ok(peer_set_change)).await?;
+                }
+                HandshakeFailed { failed_addr, error } => {
+                    if success_count <= constants::GET_ADDR_FANOUT {
+                        // this creates verbose logs, but it's better than just hanging on
+                        // startup with no output
+                        info!(addr = ?failed_addr.addr,
+                      ?error,
+                          ?success_count,
+                          ?peerset_initial_target_size,
+                      "an initial peer connection failed");
+                    } else {
+                        // switch to debug when we have enough peers
+                        debug!(addr = ?failed_addr.addr,
+                      ?error,
+                      ?success_count,
+                           ?peerset_initial_target_size,
+                           "an initial peer connection failed");
```

Comment:
```suggestion
                        info!(
                            addr = ?failed_addr.addr,
                            ?error,
                            ?success_count,
                            ?peerset_initial_target_size,
                            "an initial peer connection failed"
                        );
                    } else {
                        // switch to debug when we have enough peers
                        debug!(
                            addr = ?failed_addr.addr,
                            ?error,
                            ?success_count,
                            ?peerset_initial_target_size,
                            "an initial peer connection failed"
                       );
```

I think `rustfmt` can't format this properly, but I followed what it did for the `debug!` on line 275.

Reply:
Yeah rustfmt tends to ignore macros, because it can't be sure it will get the formatting right.

This method is totally deleted in #2160.